pkg/volume/azure_file/azure_file.go

```diff
@@ -208,15 +209,27 @@ func (b *azureFileMounter) SetUpAt(dir string, fsGroup *int64) error {
 	if accountName, accountKey, err = b.util.GetAzureCredentials(b.plugin.host, b.pod.Namespace, b.secretName); err != nil {
 		return err
 	}
-	os.MkdirAll(dir, 0700)
 
-	source := fmt.Sprintf("//%s.file.%s/%s", accountName, getStorageEndpointSuffix(b.plugin.host.GetCloudProvider()), b.shareName)
-	// parameters suggested by https://azure.microsoft.com/en-us/documentation/articles/storage-how-to-use-files-linux/
-	options := []string{fmt.Sprintf("vers=3.0,username=%s,password=%s,dir_mode=0700,file_mode=0700", accountName, accountKey)}
-	if b.readOnly {
-		options = append(options, "ro")
+	mountOptions := []string{}
+	source := ""
+	osSeparator := string(os.PathSeparator)
+	source = fmt.Sprintf("%s%s%s.file.%s%s%s", osSeparator, osSeparator, accountName, getStorageEndpointSuffix(b.plugin.host.GetCloudProvider()), osSeparator, b.shareName)
+
+	if runtime.GOOS == "windows" {
+		cmd := fmt.Sprintf(`$User = \"AZURE\\%s\";$PWord = ConvertTo-SecureString -String \"%s\" -AsPlainText -Force;
+			$Credential = New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList $User, $PWord`,
+			accountName, accountKey)
```

Comment:
I'd suggest passing 'username=%s,password=%s' only, and leave implementation to mounter.

Those directives have enforced underlying implementation, and it's for SMB protocol only.

The underlying mounter implementation is based on OS, so for windows, there's only one implementation (to mount AzureFile/AzureDisk or other kinds).

The problem is windows does not have 'mount' option semantic. We should be careful here to introduce new ones.

A potential reference is mount cmd for NFS on Windows:
https://technet.microsoft.com/en-us/library/cc754350(v=ws.11).aspx